From 2bc328059ac2eecbb6344bbaf46d78ff4c15615b Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Tue, 25 Sep 2018 13:47:47 -0700
Subject: [PATCH 01/13] DO-NOT-UPSTREAM: Disable -Wframe-larger-than= for Clang
 with KASAN

This is an LLVM issue, all of the warnings don't happen with KASAN
disabled: https://bugs.llvm.org/show_bug.cgi?id=38809

Link: https://github.com/ClangBuiltLinux/linux/issues/39
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 lib/Kconfig.debug | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index c9580134ff11..7bbff662016a 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -222,6 +222,7 @@ config ENABLE_MUST_CHECK
 config FRAME_WARN
 	int "Warn for stack frames larger than (needs gcc 4.4)"
 	range 0 8192
+	default 0 if KASAN && CC_IS_CLANG
 	default 3072 if KASAN_EXTRA
 	default 2048 if GCC_PLUGIN_LATENT_ENTROPY
 	default 1280 if (!64BIT && PARISC)
-- 
2.19.1

